frontend/src/certificate_verification.tsx

```diff
@@ -0,0 +1,266 @@
+    fill: "#388e3c", // same as success.dark in the default theme
+    marginLeft: "0.5em",
+  },
+  warnIcon: {
+    fill: "#ff9800", // same as warning.main in the default theme
+    marginLeft: "0.5em",
+  },
+  failIcon: {
+    fill: "#d32f2f", // same value as error.dark in the default theme
+    marginLeft: "0.5em",
+  },
+  boxTitle: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+  },
+});
+
+export function CertificateVerification({
+  id,
+  individual,
+}: {
+  id: string;
+  individual: Individual | undefined;
+}) {
+  const [file, setFile] = React.useState(undefined as File | undefined);
+  const [certValid, setCertValid] = React.useState(false as boolean);
+  const [certOutdated, setCertOutdated] = React.useState(false as boolean);
+  const [certNotFound, setCertNotFound] = React.useState(false as boolean);
+  const [showPdf, setShowPdf] = React.useState(false as boolean);
+  //States to make pdf-preview-library work
+  const [numPages, setNumPages] = React.useState(null);
+  const [pageNumber, setPageNumber] = React.useState(1);
+  const [previewUrl, setPreviewUrl] = React.useState(undefined as unknown);
+  const [previewWidth, setPreviewWidth] = React.useState(0 as Number);
+  const style = useStyles();
+  const { userMessage } = useMessageContext();
+
+  const verifyCertificate = (id: string) => {
+    const postCertificate = (
+      id: string,
+      content: string | ArrayBuffer | undefined
+    ) => {
+      fetch(`/api/certificates/verify/${id}`, {
+        body: content,
+        method: "POST",
+        credentials: "same-origin",
+        headers: {
+          "Content-Type": "application/pdf",
+        },
+      })
+        .then((res) => {
+          if (res.status === 200) {
+            setCertValid(true);
+          } else if (res.status === 202) {
+            setCertOutdated(true);
+          } else if (res.status === 404) {
+            setCertNotFound(true);
+          } else {
+            throw new Error("Något gick fel.");
+          }
+        })
+        .catch((error) => {
+          userMessage(error.message, "error");
+        });
+    };
+
+    if (file) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        const binaryFile = event.target?.result
+          ? event.target.result
+          : undefined;
+        postCertificate(id, binaryFile);
+      };
+      reader.readAsArrayBuffer(file);
+    }
+  };
+
+  const getLatestCertificate = (id: string) => {
+    fetch(`/api/certificates/preview/${id}`, {
+      method: "GET",
+      credentials: "same-origin",
+      headers: {
+        Accept: "application/octet-stream",
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) => res.arrayBuffer())
+      .then((data) => {
+        setPreviewWidth(window.innerWidth * 0.5);
+        setPreviewUrl(data);
+      })
+      .catch((error) => {
+        userMessage(error.message, "error");
+      });
+  };
```

Comment:
Maybe this code should be in `communication.tsx`together with all the other HTTP request handlers.